components/usage/pkg/db/workspace_instance.go

```diff
@@ -214,14 +215,26 @@ type WorkspaceInstanceForUsage struct {
 
 // WorkspaceRuntimeSeconds computes how long this WorkspaceInstance has been running.
 // If the instance is still running (no stopping time set), maxStopTime is used to to compute the duration - this is an upper bound on stop
-func (i *WorkspaceInstanceForUsage) WorkspaceRuntimeSeconds(maxStopTime time.Time) int64 {
+func (i *WorkspaceInstanceForUsage) WorkspaceRuntimeSeconds(stopTimeIfInstanceIsStillRunning time.Time) int64 {
 	start := i.StartedTime.Time()
-	stop := maxStopTime
+	stop := stopTimeIfInstanceIsStillRunning
 
 	if i.StoppingTime.IsSet() {
-		if i.StoppingTime.Time().Before(maxStopTime) {
```

Comment:
This was really the problem. If we wanted to retain the logic, we'd have needed to do 
```
if i.StoppingTime.Time().After(maxStopTime) {
```